drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,458 @@
+ * @brief Writes data to device register (I2C)
+ * @param dev 		  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param bytes 	  - Number of bytes to write from starting register address
+ * @param write_data  - Write data buffer
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_write(struct max9611_dev *dev, enum max9611_regs addr,
+		  uint8_t bytes, uint8_t *write_data)
+{
+	int i;
+	uint8_t buf[bytes + 1];
+
+	/* Avoid write to out of bounds register */
+	if (addr + bytes > 0x0C)
+		return -EINVAL;
+
+	/* Exit when writing to Read only registers */
+	if (addr < MAX9611_CTR1)
+		return -EINVAL;
+
+	buf[0] = addr;
+
+	for (i = 0; i < bytes; i++)
+		buf[i + 1] = write_data[i];
+
+	return no_os_i2c_write(dev->i2c_desc, buf, bytes + 1, 1);
+}
+
+/***************************************************************************//**
+ * @brief Update selected I2C register contents
+ * @param dev   	  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param update_val  - Value used to update on selected register
+ * @param update_mask - Mask to update to selected register bit/s
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_reg_update(struct max9611_dev *dev, enum max9611_regs addr,
+		       int update_val, int update_mask)
+{
+	uint8_t temp[1];
+	int ret;
+
+	ret = max9611_read(dev, addr, 1, temp);
+	if (ret)
+		return ret;
+
+	temp[0] &= ~update_mask;
+	temp[0] |= update_val;
+
+	return max9611_write(dev, addr, 1, temp);
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 operating mode
+ * @param dev 		  - The device structure.
+ * @param mode 		  - MODE configuration
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_mode(struct max9611_dev *dev, enum max9611_mode_conf mode)
+{
+	mode <<= no_os_find_first_set_bit(MAX9611_MODE_MASK);
+
+	return max9611_reg_update(dev, MAX9611_CTR1, mode,
+				  MAX9611_MODE_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get the MAX9611 operating mode
+ * @param dev 		  - The device structure.
+ * @param mode 		  - Pointer to store the obtained mode
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_mode(struct max9611_dev *dev, enum max9611_mode_conf *mode)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR1, 1, &temp);
+	if (ret)
+		return ret;
+
+	*mode = (temp & MAX9611_MODE_MASK) >> no_os_find_first_set_bit(
```

Comment:
use `no_os_field_prep` / `no_os_field_get`

Reply:
Resolved in latest commit.